src/spi.rs

```diff
@@ -102,6 +102,9 @@ pub mod config {
     pub struct Config {
         pub baudrate: Hertz,
         pub data_mode: embedded_hal::spi::Mode,
+        pub no_dummy: bool,
+        pub dma_channel: u32,
+        pub max_transfer_size: i32,
```

Comment:
`i32`? I think we should be using `usize` here.
However the real issue is that the SPI driver code is heavily utilizing [this](https://github.com/esp-rs/esp-idf-hal/blob/master/src/spi.rs#L43), which means that all transfers - including new shiny DMA - are constrained to multiple transactions of 64 bytes each. What is the point of using DMA in the first place, if we'll use it for small chunks of 64 bytes?

I think the PR must be more comprehensive, in that it should inspect all code-paths, and lift the chunking of 64 bytes restriction where possible. Note that this would not be that easy when your read and write buffers are both non-null and of a different size, but should be still quite possible. And would also allow us to get rid of these on-stack allocated buffers of `TRANS_LEN`, which seem unnecessary.